edgedb-client/src/builder.rs

```diff
@@ -1110,27 +1212,60 @@ impl Builder {
                 Run `edgedb project init` or use environment variables \
                 to configure connection."));
         }
+        let host = self.get_host();
+        if host.is_empty() {
+            return Err(ClientError::with_message("invalid host: empty string"));
+        } else if host.contains(",") {
+            return Err(ClientError::with_message("invalid host: multiple hosts"));
+        }
+        if self.get_user().is_empty() {
+            return Err(ClientError::with_message("invalid user: empty string"));
+        }
+        if self.get_database().is_empty() {
+            return Err(ClientError::with_message("invalid database: empty string"));
+        }
+        if self.get_port() == 0 {
+            return Err(ClientError::with_message("invalid port: 0"));
+        }
+        let tls_security;
         let verifier = match self.tls_security {
-            _ if self.insecure() => Arc::new(tls::NullVerifier) as Verifier,
-            Insecure => Arc::new(tls::NullVerifier) as Verifier,
+            _ if self.insecure() => {
+                match self.tls_security {
```

Comment:
why do you put match of the same thing inside the match? Please move these braches to the outer match.

Reply:
After reviewing this, I think there was an inconsistancy issue in `insecure_dev_mode`. Fixed [here](https://github.com/edgedb/edgedb-cli/pull/905/commits/ec7c1af6fc339e40fdfb2f42940541b36779bd6f), please kindly double check.

Reply:
Looks good. Thanks!